### ISSUE TITLE
three shaking / look into dependencies

### ISSUE THREAD

**Issue by Mael-Abgrall** [A-star-logic/memoire#10]
Some dependencies have an ungodly size, it might be interesting to look into them, remove or replace the bloated files:

```bash
# result of: du -h --max-depth=2 | sort -h
7.1M    ./node_modules/langchain
9.8M    ./node_modules/@esbuild
13M     ./node_modules/@langchain
13M     ./node_modules/@sentry
22M     ./node_modules/js-tiktoken
23M     ./node_modules/tiktoken
24M     ./node_modules/@opentelemetry
```

**Comment by Mael-Abgrall** [comment]
Some notes & findings (after taking too much time on this):

- the alpine image of node doesn't work with filesystem calls &/or might require a deeper look. This would save 66mb -> not worth it for now
- Using Rollup without bundling node_modules result in a much smaller package (about a hundred Kbs compared to 1.1Mb with tsup); This could be useful in the future, but since 99% of the size of the app is due to node_modules, this doesn't make a big difference today (although it might be interesting when the app gets bigger, as it also bundle everything in one or two files, making startup slightly faster too)
- Using Rollup while bundling node_modules result in a much, much smaller app footprint (between 10 and 18Mb). However there was a lot of unresolved issues around Cjs (even with noderesolver and commonjs plugins), the main culprit being opentelemetry, which is unfortunately used by Sentry (we can't remove Sentry)

tldr:
- We save between 100 and 150mb by bundling everything with rollup (or up to 200mb with alpine), but this would require a lot of work to understand how to make the bundle work; it's not worth the time today since the app already takes only 330mb.

next steps:

- Removing langchain will result in a lot of free space saving, we use it only for the local model (but not required to keep using it; can replace the library, or use an online model); and for the text splitter -> but similarly we need a smarter text splitter which does not require langchain
- Need to understand more why tiktoken take so much space, it seems there are smaller wasm files available, but need to figure out how to make it work
